turbopack/crates/turbo-tasks-fs/src/read_glob.rs

```diff
@@ -53,32 +42,97 @@ async fn read_glob_internal(
                 let entry = entry.resolve_symlink().await?;
                 match entry {
                     DirectoryEntry::Directory(path) => {
-                        let full_path = format!("{prefix}{segment}");
-                        let full_path_prefix: RcStr = format!("{full_path}/").into();
-                        if glob_value.execute(&full_path) {
+                        let full_path = &path.await?.path;
```

Comment:
`"{prefix}{segment}"` is not always the same as `path.await?.path`.
Yes in most cases it's the same, but not when there are symlinks in the path. The glob need to match the unresolved path, while the `path` will be the resolved location.

Reply:
good catch

Added some tests and reproduced.

Reverted back to the old prefix accumulation mode since that appears to be how this was handled before, we can track our path to the file and still follow symlinks.

Also fixed anopther issue involving symlink cycles.